tape/src/main/java/com/squareup/tape/QueueFile.java

```diff
@@ -519,26 +519,42 @@ public synchronized int size() {
    * @throws java.util.NoSuchElementException if the queue is empty
    */
   public synchronized void remove() throws IOException {
+    remove(1);
+  }
+
+  /**
+   * Removes the eldest {@code n} elements.
+   *
+   * @throws java.util.NoSuchElementException if the queue is empty
+   */
+  public synchronized void remove(int n) throws IOException {
     if (isEmpty()) throw new NoSuchElementException();
-    if (elementCount == 1) {
+    if (n > elementCount) {
+      throw new IllegalArgumentException(
+          "Cannot remove more elements (" + n + ") than present in queue (" + elementCount + ".");
+    } else if (n == elementCount) {
```

Comment:
This is going to be stupid nitpicky, but I think this method reads better like this:

```
if (empty) throw;
if (n > count) {
  throw;
}
if (n == count) {
  clear();
  return;
}

normal multi-remove
```
This way there's no cognitive load with the conditionals. Two state checks, one optimization check, and then the normal method body. I never have to think about `else` or `elseif` and what was already checked.